comictalker/talkers/comicvine.py

```diff
@@ -0,0 +1,759 @@
+            cv_response = self.get_cv_content(urljoin(self.api_base_url, "issues/"), params)
+
+            volume_issues_result.extend(cast(list[CVIssueDetailResults], cv_response["results"]))
+            current_result_count += cv_response["number_of_page_results"]
+
+        # Format to expected output !! issues/ volume does NOT return publisher!!
+        formatted_volume_issues_result = self.format_issue_results(volume_issues_result)
+
+        cvc.add_volume_issues_info(self.source_name, formatted_volume_issues_result)
+
+        return formatted_volume_issues_result
+
+    def fetch_issues_by_volume_issue_num_and_year(
+        self, volume_id_list: list[int], issue_number: str, year: str | int | None
+    ) -> list[ComicIssue]:
+        volume_filter = ""
+        for vid in volume_id_list:
+            volume_filter += str(vid) + "|"
+        flt = f"volume:{volume_filter},issue_number:{issue_number}"
+
+        int_year = utils.xlate(year, True)
+        if int_year is not None:
+            flt += f",cover_date:{int_year}-1-1|{int_year + 1}-1-1"
+
+        params: dict[str, str | int] = {
+            "api_key": self.api_key,
+            "format": "json",
+            "field_list": "id,volume,issue_number,name,image,cover_date,site_detail_url,description,aliases,associated_images",
+            "filter": flt,
+        }
+
+        cv_response = self.get_cv_content(urljoin(self.api_base_url, "issues/"), params)
+
+        current_result_count = cv_response["number_of_page_results"]
+        total_result_count = cv_response["number_of_total_results"]
+
+        filtered_issues_result = cast(list[CVIssueDetailResults], cv_response["results"])
+        page = 1
+        offset = 0
+
+        # see if we need to keep asking for more pages...
+        while current_result_count < total_result_count:
+            page += 1
+            offset += cv_response["number_of_page_results"]
+
+            params["offset"] = offset
+            cv_response = self.get_cv_content(urljoin(self.api_base_url, "issues/"), params)
+
+            filtered_issues_result.extend(cast(list[CVIssueDetailResults], cv_response["results"]))
+            current_result_count += cv_response["number_of_page_results"]
+
+        formatted_filtered_issues_result = self.format_issue_results(filtered_issues_result)
+
+        return formatted_filtered_issues_result
+
+    def fetch_issue_data(self, series_id: int, issue_number: str) -> GenericMetadata:
+        issues_list_results = self.fetch_issues_by_volume(series_id)
+
+        # TODO Is this required?
+        f_record = None
+        for record in issues_list_results:
+            if not IssueString(issue_number).as_string():
+                issue_number = "1"
+            if (
+                IssueString(record["issue_number"]).as_string().casefold()
+                == IssueString(issue_number).as_string().casefold()
+            ):
+                f_record = record
+                break
```

Comment:
This is a result of there being no cache function to return a record based on series id and issue number. Should I leave this as it is or put in a cache function to retrieve a record based on series id and issue number?

Reply:
`fetch_issues_by_volume` is already cached and then you can replace lines 701-716 with a call to `fetch_issue_data_by_issue_id`

Reply:
Yep, that's true for getting the issue data. I'll change that. 

My question was about looping through the issue list rather than the cache returning 1 record (if it's found).

Looking closer, with the changes to use the issue id instead, this should never get used anyway and is only a failover, so there's no need to try and optimize it.